docs/examples/sensormanagement/bayesian_search_example_2.py

```diff
@@ -0,0 +1,465 @@
+ny = 15
+
+xarray = np.linspace(xmin, xmax, nx)
+yarray = np.linspace(ymin, ymax, ny)
+x, y = np.meshgrid(xarray, yarray)
+
+x_pos = x.flatten()
+y_pos = y.flatten()
+
+# %%
+# In setting the particle weights here, we are defining our initial probability distribution. In
+# this case, the initial distribution will be Gaussian, centred around (x=7, y=8).
+
+
+from stonesoup.types.state import ParticleState
+from stonesoup.types.array import StateVectors
+from scipy.stats import multivariate_normal
+
+# parameterise our prior probability distribution
+mu = np.array([7, 8])  # mean(x, y)
+sigma = np.array([[10, 0], [3, 6]])  # covariance matrix [[var(x), covar(xy)], [covar(xy), var(y)]]
+
+prior = multivariate_normal.pdf(np.vstack((x_pos, y_pos)).T, mu, sigma)
+
+prior = prior / sum(prior)
+prior_weights = prior.reshape(ny, nx).flatten()
+
+vel = [0] * len(x_pos)
+state_vectors = StateVectors(np.array([x_pos, vel, y_pos, vel]))
+
+prior = ParticleState(state_vector=state_vectors,
+                      weight=prior_weights,
+                      timestamp=start_time)
+
+
+# %%
+# Creating a Custom Reward Function
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# At each timestep, we want our platform to move to the area with the greatest probability of
+# containing the target. We create a custom reward function with identical functionality to that
+# in the first example.
+
+
+from stonesoup.sensor.sensor import Sensor
+
+
+def sumofweightsreward(config, undetectmap, timestamp):
+    """
+    Takes a configuration, which is a mapping of actionables (sensors, platforms, etc.) to their
+    possible actions. For each item in the mapping, creates a copy of the actionable which is used
+    to simulate the action (e.g., platform movement). Then takes a measurement with the simulated
+    sensors to count the number of particles that would be in the sensor's FOV if the action
+    were taken. Returns sum of particles within the sensor's FOV.
+    """
+    predicted_sensors = set()
+    memo = {}
+    for actionable, actions in config.items():
+        predicted_actionable = copy.deepcopy(actionable, memo)
+        predicted_actionable.add_actions(actions)
+        predicted_actionable.act(timestamp)
+
+        if isinstance(predicted_actionable, Sensor):
+            predicted_sensors.add(predicted_actionable)
+
+    running_tot = 0
+    for sensor in predicted_sensors:
+
+        # assume no detection
+        for j, particle in enumerate(undetectmap.state_vector):
+            pstate = ParticleState(particle)
+            weight = undetectmap.weight[j]
+            # add to running total if in FOV
+            if sensor.is_detectable(pstate):
+                running_tot += weight
+
+    return float(running_tot)
+
+
+# %%
+# Creating a Sensor Manager
+# ~~~~~~~~~~~~~~~~~~~~~~~~~
+# Finally, we create a :class:`~.BruteForceSensorManager`, which will evaluate every possible
+# movement of our platform using the custom reward function and return the best one.
+
```

Comment:
'Best' is subjective -- it will return the action with the highest numerical reward value.